### PR TITLE
Update UG for CS2101

### PR DESCRIPTION
**Pull request by brelkh**
1. Added intro and purpose of UG (to cover front matters in UG rubrics)
2. Added intro to some sections to help readers a bit more.
3. Edited some instructions to have more "you" language. Supposed to be more warm and friendly.
4. Added some FAQs.
5. Added screenshots for `history`, `undo` and `clear` commands.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -18,7 +24,7 @@ especially for those who are teaching multiple classes/modules at the same time.
 
 ## Quick start
 
-1. Ensure you have [Java `11`](https://www.oracle.com/java/technologies/downloads/#java11) or above installed in your Computer.
+1. Before getting started with TAPA, ensure you have [Java `11`](https://www.oracle.com/java/technologies/downloads/#java11) or above installed on your computer. You can check the version of Java on your computer by inputting `java -version` in your terminal.
```
Comment (Benjykoh) [inline docs/UserGuide.md:27]: Maybe in this part, we can add and say like how to go to your terminal? I've seen some guides out there that actually specify this so maybe we can do it as well to make it as beginner-friendly as possible?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -66,39 +72,44 @@ Each contact in TAPA represents a student, who can have the following fields:
...
+* In TAPA, two students are considered the same only if they have the same student ID (i.e. multiple students can have the same module code/name/phone number/telegram handle/email address but not the same student IDs).
 
 * More details regarding the use of prefix can be found in the [Features](#features) section below.
 </div>
 
 
 ## Features
 
+**Introduction**<br>
+TAPA offers a plethora of features at your disposal. However, you need to use different command formats to execute the various features. Before diving in to TAPA's commands and features, do read the following notes to understand how to format your command inputs to TAPA:
```
Comment (Benjykoh) [inline docs/UserGuide.md:86]:
```suggestion
TAPA offers a plethora of features at your disposal. However, you need to use different command formats to execute the various features. Before diving into TAPA's commands and features, do read the following notes to understand how to format your command inputs to TAPA:
```

**Review by Benjykoh — COMMENTED** [review]
Almost LGTM! Got some minor comments but overall a great gob! Thanks for taking the time to make these edits :)

File: docs/UserGuide.md (hunk 1 above)
Comment (brelkh) [reply docs/UserGuide.md:27]: Got it! Maybe I'll just link to an article that talks about that?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -18,7 +24,7 @@ especially for those who are teaching multiple classes/modules at the same time.
 
 ## Quick start
 
-1. Ensure you have [Java `11`](https://www.oracle.com/java/technologies/downloads/#java11) or above installed in your Computer.
+1. Before getting started with TAPA, ensure you have [Java `11`](https://www.oracle.com/java/technologies/downloads/#java11) or above installed on your computer. You can check the version of Java on your computer by inputting `java -version` in your [terminal](https://www.ionos.com/help/email/troubleshooting-mail-basicmail-business/access-the-command-prompt-or-terminal/#:~:text=Open%20Command%20Prompt%20in%20Windows,cmd%22%20and%20then%20click%20OK.).
```
Comment (brelkh) [inline docs/UserGuide.md:27]: @Benjykoh Do you think this link will help a reader access terminal?

Comment (Benjykoh) [reply docs/UserGuide.md:27]: @brelkh Yup I think this would be great!

**Review by Benjykoh — APPROVED** [review]
LGTM!

**Comment by brelkh** [comment]
@Benjykoh thanks for the review!

**Pull request merged**